src/shared/metadata/selectors.js

```diff
@@ -412,22 +416,46 @@ const resolveCategoryOptionIds = (categories, categoryOptions) => {
 
 export const getCategoriesWithOptionsWithinPeriod = createCachedSelector(
     (metadata) => metadata,
-    (_, dataSetId) => dataSetId,
+    getDataSetById,
     (_, __, periodId) => periodId,
-    (metadata, dataSetId, periodId) => {
-        if (!dataSetId || !periodId) {
+    (metadata, dataSet, periodId) => {
+        if (!dataSet?.id || !periodId) {
             return []
         }
 
-        const relevantCategories = getCategoriesByDataSetId(metadata, dataSetId)
+        const relevantCategories = getCategoriesByDataSetId(
+            metadata,
+            dataSet?.id
+        )
+
         const categoryOptions = getCategoryOptions(metadata)
         const relevantCategoriesWithOptions = resolveCategoryOptionIds(
             relevantCategories,
             categoryOptions
         )
         const period = parsePeriodId(periodId)
         const periodStartDate = new Date(period.startDate)
-        const periodEndDate = new Date(period.endDate)
+
+        // periodEndDate is up to following start date
+        let periodEndDate = addFullPeriodTimeToDate(
+            periodStartDate,
+            period?.periodType?.type
+        )
+
+        // reduce perfiodEndDate by openPeriodsAfterCoEndDate
+        const openPeriodsAfterCoEndDate = Math.max(
+            dataSet?.openPeriodsAfterCoEndDate || 0,
+            0
+        )
+        for (let i = 0; i < openPeriodsAfterCoEndDate; i++) {
```

Comment:
Would it make sense to implement a `removePeriodsTimeToDate`-function in `fixed-periods.js`?

`const removePeriodsTimeToDate = periods => createUpdateFullPeriodTimeToDate(-1 * periods).`